src/util-hash.c

```diff
@@ -138,44 +154,27 @@ int HashTableAdd(HashTable *ht, void *data, uint16_t datalen)
         SCFree(hb);
     return -1;
 }
-
+/**
+ *  \brief Remove an item from the hash table
+ *  \details This function will search for the item in the hash table and remove it if found
+ *  \note If the Free function is set, it will be called for the data item being removed.
+ *  \param ht Pointer to the HashTable
+ *  \param data Pointer to the data to remove
+ *  \param datalen Length of the data to remove
+ *  \return int 0 on success, -1 if the item was not found or an error occurred
+ */
 int HashTableRemove(HashTable *ht, void *data, uint16_t datalen)
 {
     uint32_t hash = ht->Hash(ht, data, datalen);
 
-    if (ht->array[hash] == NULL) {
-        return -1;
-    }
-
-    if (ht->array[hash]->next == NULL) {
```

Comment:
Would not the fix be simply remove this `if` block ?

Or to add the condition like `if (ht->Compare(hashbucket->data,hashbucket->size,data,datalen) == 1)` to run it ?

Reply:
Yeah sure, I should have mentioned it